### PR TITLE
chore: add earn routes

### PR DESCRIPTION
**Pull request by NguyenHuy1812**
**What does this PR do?**

-   [x] Add earn routes

### CODE REVIEW

File: pkg/routes/v1.go [hunk 1]
```diff
@@ -570,4 +577,10 @@ func NewRoutes(r *gin.Engine, h *handler.Handler, cfg config.Config) {
 	{
 		emojisGroup.GET("", h.Emojis.ListEmojis)
 	}
+
+	earnGroup := v1.Group("earn")
```
Comment (trkhoi) [inline pkg/routes/v1.go:581]: /earn

File: pkg/repo/user_earn/pg.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	return userEarn, tx.Commit().Error
+}
+
+func (pg *pg) Delete(userEarn *model.UserEarn) (*model.UserEarn, error) {
+	return userEarn, pg.db.Where("user_id = ? and earn_id = ?", userEarn.UserId, userEarn.EarnId).Delete(userEarn).Error
+}
+
+func (pg *pg) GetByUserId(q ListQuery) ([]model.UserEarn, int64, error) {
```
Comment (trkhoi) [inline pkg/repo/user_earn/pg.go:37]: func (pg *pg) GetByUserId(q ListQuery) (userEarns []model.UserEarn, total int64, err error) {
...
}

File: pkg/repo/earn_info/pg.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+func (pg *pg) Create(earnInfo *model.EarnInfo) (*model.EarnInfo, error) {
+	return earnInfo, pg.db.Create(earnInfo).Error
+}
+
+func (pg *pg) GetById(id int64) (earnInfo *model.EarnInfo, err error) {
+	return earnInfo, pg.db.First(&earnInfo, id).Error
+}
+
+func (pg *pg) List(q ListQuery) ([]model.EarnInfo, int64, error) {
```
Comment (trkhoi) [inline pkg/repo/earn_info/pg.go:25]: func (pg *pg) List(q ListQuery) (earnInfos []model.EarnInfo, total int64, err error) {
...
}

File: pkg/handler/earn/earn.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+// @Accept      json
+// @Produce     json
+// @Param       page   query  string false  "page"
+// @Param       size   query  string false  "size"
+// @Success     200 {object} response.EarnInfoListResponse
+// @Router      /earn [get]
+func (h *Handler) GetEarnInfoList(c *gin.Context) {
+	page := c.Query("page")
+	size := c.Query("size")
+	if page == "" {
```
Comment (trkhoi) [inline pkg/handler/earn/earn.go:41]: a có thể tạo 1 model request r để default vô cho nó
example:
```
type PaginationRequest struct {
	Page  int64  `json:"page" form:"page,default=0"`
	Size  int64  `json:"size" form:"size,default=10"`
}

req := request.PaginationRequest{}
	if err := c.ShouldBindQuery(&req); err != nil {
		h.log.Error(err, "[handler.GetEarnInfoList] BindJSON() failed")
		c.JSON(http.StatusBadRequest, response.CreateResponse[any](nil, nil, err, nil))
		return
	}
```

File: pkg/handler/earn/earn.go [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+// @Router      /users/{id}/earn [get]
+func (h *Handler) GetUserEarnListByUserId(c *gin.Context) {
+	userId := c.Param("id")
+	if userId == "" {
+		h.log.Info("[handler.GetUserEarnListByUserId] - user id empty")
+		c.JSON(http.StatusBadRequest, response.CreateResponse[any](nil, nil, errors.New("user_id is required"), nil))
+		return
+	}
+
+	page := c.Query("page")
```
Comment (trkhoi) [inline pkg/handler/earn/earn.go:143]: same as above and should put userId into struct too

File: pkg/entities/earn.go [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+	if err != nil {
+		e.log.Fields(logger.Fields{"req": req}).Errorf(err, "[entity.CreateEarnInfo] - e.repo.EarnInfo.Create failed")
+		return nil, err
+	}
+
+	return earn, nil
+}
+
+func (e *Entity) GetEarnInfoList(page, size string) (*response.EarnInfoListResponse, error) {
+	pageInt, _ := strconv.Atoi(page)
```
Comment (trkhoi) [inline pkg/entities/earn.go:38]: should handle in handler, create validation for it

File: pkg/handler/earn/earn.go [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+// @Router      /users/{id}/earn/{earn_id} [delete]
+func (h *Handler) DeleteUserEarn(c *gin.Context) {
+	userId := c.Param("id")
+	if userId == "" {
+		h.log.Info("[handler.GetUserEarnListByUserId] - user id empty")
+		c.JSON(http.StatusBadRequest, response.CreateResponse[any](nil, nil, errors.New("user_id is required"), nil))
+		return
+	}
+
+	earnIdStr := c.Param("earn_id")
```
Comment (trkhoi) [inline pkg/handler/earn/earn.go:181]: use struct same as above

**Review by trkhoi — CHANGES_REQUESTED** [review]

**Review by trkhoi — APPROVED** [review]

**Review by quanghuynguyen1902 — APPROVED** [review]

**Review by vhbien000 — APPROVED** [review]

**Pull request merged**